arroyo-api/src/main.rs

```diff
@@ -750,6 +746,101 @@ impl ApiGrpc for ApiServer {
         Ok(Response::new(DeleteJobResp {}))
     }
 
+    async fn refresh_sample(
+        &self,
+        request: Request<RefreshSampleReq>,
+    ) -> Result<Response<RefreshSampleResp>, Status> {
+        let (request, auth) = self.authenticate(request).await?;
+        // start the pipeline associated with the source
+        let job_id = api_queries::get_source_pipeline() // TODO: when it's possible to have two jobs for one pipeline, how does this look?
+            .bind(
+                &self.client().await?,
+                &(request.into_inner().source_id as i64),
+            )
+            .one()
+            .await
+            .map_err(log_and_map)?;
+
+        self.update_job(Request::new(UpdateJobReq {
+            job_id: job_id.clone(),
+            stop: Some(StopType::None.into()),
+            checkpoint_interval_micros: None,
+            parallelism: None,
+        }))
+        .await?;
+
+        // subscribe to the output
+        let details = get_job_details(&job_id, &auth, &self.client().await?).await?;
+        if !details
+            .job_graph
+            .unwrap()
+            .nodes
+            .iter()
+            .any(|n| n.operator.contains("GrpcSink"))
+        {
+            // TODO: make this check more robust
+            return Err(Status::invalid_argument(format!(
+                "Job {} does not have a web sink",
+                job_id
+            )));
+        }
+
+        let mut controller = ControllerGrpcClient::connect(self.controller_addr.clone())
+            .await
+            .map_err(log_and_map)?;
+
+        info!("connected to controller");
+
+        let mut stream = controller
+            .subscribe_to_output(Request::new(grpc::GrpcOutputSubscription {
+                job_id: job_id.clone(),
+            }))
+            .await
+            .map_err(log_and_map)?
+            .into_inner();
```

Comment:
TODO: this subscription logic is similar to `subscribe_to_output` function.  Could be combined into a single function that is called by both?